torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -204,15 +213,24 @@ PyRRef pyRemotePythonUdf(
       SerializedPyObj(std::move(pickledPythonUDF), std::move(tensors));
   if (ctx.getWorkerId() != dst.id_) {
     auto userRRef = ctx.createUserRRef(dst.id_, PyObjectType::get());
-    ctx.addPendingUser(userRRef->forkId(), userRRef);
     auto fm = sendPythonRemoteCall(
         dst,
         std::move(serializedPyObj),
         userRRef->rrefId().toIValue(),
         userRRef->forkId().toIValue(),
         rf);
 
-    fm->addCallback(callback::confirmPendingUser);
+    ctx.addPendingUser(userRRef->forkId(), userRRef);
+    fm->addCallback([forkId{userRRef->forkId()}](
```

Comment:
This looks the same as the the other two occurrences. Any reason for not directly modify `confirmPendingUser`?

Reply:
This was a symptom of an evolving review. You're right - I just restored the ``confirmPendingUser()`` function, albeit with a new ``expectedForkId`` arg

I had to wrap it with a lambda to bind the expectedForkId arg, but looks cleaner with the suggestion.